src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -777,6 +775,66 @@ public int EnsureCapacity(int capacity)
             return newSize;
         }
 
+        /// <summary>
+        /// Sets the capacity of this dictionary to hold up at least all existing entries
+        /// 
+        /// This method can be used to minimize the memory overhead 
+        /// once it is known that no new elements will be added. 
+        /// 
+        /// To completely clear a dictionary and release all memory 
+        /// referenced by the dictionary, execute the following statements:
+        /// 
+        /// dictionary.Clear();
+        /// dictionary.TrimExcess();
+        /// </summary>
+        public void TrimExcess()
+        {
+            TrimExcess(HashHelpers.GetPrime(Count));
+        }
+
+        /// <summary>
+        /// Sets the capacity of this dictionary to hold up at least 'capacity' entries without any further expansion of its backing storage
+        /// </summary>
+        public void TrimExcess(int capacity)
+        {
+            if (capacity < Count)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.capacity);
```

Comment:
We should try to start including the invalid value in argument exceptions so that callstacks are more actionable. AOORE already has a constructor `public ArgumentOutOfRangeException(String paramName, Object actualValue, String message)` that we don't seem to use. If you want you could add an overload in ThrowHelper so that you could call this one so you can pass in the value of capacity.